### DISCUSSION TITLE
system.notsupportedexception

### DISCUSSION THREAD

**Discussion by Unix-Kernel** [Tyrrrz/DiscordChatExporter#612] [Q&A]
The application will not install in Windows 7x64-SP1 with runtime 3.1 and all rollup updates.
When trying to install the application, it errors with the following message which is related to TLS 1.2:

system.notsupportedexception: the requested security protocol not supported
at System.Net.ServicePointManager.set_SecurityProtool(SecurityProtocolType value) at DotnetRutimeboostrapper.Executable.Program.init()
at DotnetRuntimeBootstrapper.Executable.Program.Main(String[]args)

![notsupportedexception](https://user-images.githubusercontent.com/61941051/122579179-a7f13380-d022-11eb-9a6a-b56ab7a00bac.png)

**Comment by Tyrrrz** [comment]
Please see https://github.com/Tyrrrz/DiscordChatExporter/discussions/610

**Reply by Unix-Kernel** [reply]
![DCE-CLI](https://user-images.githubusercontent.com/61941051/122580617-24d0dd00-d024-11eb-9c9a-c87fb4008b96.png)

**Reply by Unix-Kernel** [reply]
After installing further updates and the two dependencies from the screenshot, DCE continues to complain about the two missing runtime components which are already installed. After re-installing them, rebooting and trying to run DCE, it shows the same message again.
![dependencies](https://user-images.githubusercontent.com/61941051/122584831-d07c2c00-d028-11eb-86ed-5c3b66e8ce00.png)

Reminds me of "linux dependency hell" from back in the days...lol :)

**Reply by Unix-Kernel** [reply]
The problem with running any previous version of DCE is that it updates it to the latest version of 2.27 which will not work the next time it is opened, which means that the "auto-aupdates" must be disabled immediately. I finally got it working by backstepping to 2.25 and disabling "auto-updates".

![DCE 2 25](https://user-images.githubusercontent.com/61941051/122587948-7a10ec80-d02c-11eb-80f5-cfc3c32711d2.png)
:
